src/Draco.Compiler/Internal/Binding/Binder_UntypedExpression.cs

```diff
@@ -115,7 +117,9 @@ static string ComputeCutoff(StringExpressionSyntax str)
 
     private UntypedExpression BindNameExpression(NameExpressionSyntax syntax, ConstraintSolver constraints, DiagnosticBag diagnostics)
     {
-        var symbol = this.LookupValueSymbol(syntax.Name.Text, syntax, diagnostics);
+        Symbol symbol;
+        if (syntax.Parent is CallExpressionSyntax || (syntax.Parent is GenericExpressionSyntax && syntax.Parent?.Parent is CallExpressionSyntax)) symbol = this.LookupFunctionSymbol(syntax.Name.Text, syntax, diagnostics);
```

Comment:
What we are trying to express here is that "if this is in a context we don't wanna look up types, just look up non-type symbols", right? If so, factor out this logic please of determining if this is such a place. Something like `SyntaxMustNotReferenceTypes`. With that, we could also rename the lookup variant and binder facts predicates to reflect their roles a bit better.